astropy/io/fits/hdu/hdulist.py

```diff
@@ -1334,7 +1334,7 @@ def _flush_resize(self):
                 # flushing (on Windows--again, this is no problem on Linux).
                 for idx, mmap, arr in mmaps:
                     if mmap is not None:
-                        arr.data = self[idx].data.data
+                        arr = self[idx].data
```

Comment:
Yeah, this will basically break the purpose of this code in the first place (in which case it might as well be deleted entirely).

On one hand, the existing code is maybe trying to be too clever; on the other hand it is solving a real problem.  It would be less of a problem of there were some way to "close" a Numpy array, such that a reasonable exception would be raised if trying to access data from an array whose underlying buffer no longer exists.  That would be a good Numpy feature.  But in the meantime it's actually IMO fine to replace an ndarray's buffer with a different buffer of the same length.

I still don't know why Numpy has decided to deprecate this, but if they insist then I still think a little helper function written in Cython to update an `ndarray` object's buffer is in order...